controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -498,49 +519,44 @@ func (r *KubeadmControlPlaneReconciler) generateMachine(ctx context.Context, kcp
 	return nil
 }
 
-func generateKubeadmControlPlaneSelector(clusterName string) *metav1.LabelSelector {
-	return &metav1.LabelSelector{
-		MatchLabels: generateKubeadmControlPlaneLabels(clusterName),
-	}
-}
-
-func generateKubeadmControlPlaneLabels(clusterName string) map[string]string {
-	return map[string]string{
-		clusterv1.ClusterLabelName:             clusterName,
-		clusterv1.MachineControlPlaneLabelName: "",
-	}
-}
 
 // reconcileDelete handles KubeadmControlPlane deletion.
 // The implementation does not take non-control plane workloads into
 // consideration. This may or may not change in the future. Please see
 // https://github.com/kubernetes-sigs/cluster-api/issues/2064
 func (r *KubeadmControlPlaneReconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Cluster, kcp *controlplanev1.KubeadmControlPlane, logger logr.Logger) (_ ctrl.Result, reterr error) {
 	// Fetch Machines
-	allMachines, err := util.GetMachinesForCluster(ctx, r.Client, cluster)
+	clusterKey := types.NamespacedName{
+		Namespace: cluster.Namespace,
+		Name:      cluster.Name,
+	}
```

Comment:
We're doing this frequently enough, I'm wondering if we should consider having a helper method for it, where the helper method would take any `runtime.Object` and return a `types.NamespacedName` for that resource.

Reply:
@chuckha added a helper method